### ISSUE TITLE
Renderpass page

### ISSUE THREAD

**Issue by charles-lunarg** [KhronosGroup/Vulkan-Guide#17]
Looking at the list of topics, it seems there is room for a short page on renderpasses. 
Nothing extravagant, just a "heres what, why, when, and links to detailed descriptions" kind of thing.
I am willing to write it if others think its a useful/needed page. 

Content Proposal: 
 * Purpose & rational 
 * Simple example/use case
 * Subpasses, their limitations, and when to use them.
 * Mention how mobile uses renderpasses for efficiency
 * Links to external sources - Are these good links and what else is there to add?
    * Vulkan-Samples https://github.com/KhronosGroup/Vulkan-Samples/blob/master/samples/performance/render_subpasses/render_subpasses_tutorial.md 
    * GPUinfo https://gpuopen.com/vulkan-renderpasses/
    * Samsung https://developer.samsung.com/game/renderpasses

**Comment by Andreyogld3d** [comment]
Example for VK_KHR_create_renderpass2 and VK_KHR_depth_stencil_resolve:
https://github.com/Andreyogld3d/Vulkan_VK_KHR_depth_stencil_resolve

**Comment by marty-johnson59** [comment]
Note, this has evolved a lot recently - would be good to clarify how folks should be using render passes today - perhaps part of the "how to Vulkan in 2024" chapter?  Seems like an important topic - labeling as priority..

**Comment by SaschaWillems** [comment]
Agree. People new to Vulkan will probably be confused as to whether they should use render passes or dynamic rendering. And that has a lot of implications on how you write code, so people are sometimes a bit upset if they find out that Vulkan supports a renderpass-less approach after they wrote code using render passes.

**Comment by fluppeteer** [comment]
Asking internally in IMG, although we could historically have had some benefit from using renderpasses, we've had to throw away most of the benefits we could have had in our driver (because of things like update after bind) - so it's no longer so detrimental to use dynamic rendering (I hope). I think Qualcomm raised an issue at one point, though, so I'll try reaching out.
